gensim/models/hdpmodel.py

```diff
@@ -49,20 +49,36 @@
 
 def dirichlet_expectation(alpha):
     """
-    For a vector theta ~ Dir(alpha), compute E[log(theta)] given alpha.
+    For a vector `theta~Dir(alpha)`, compute `E[log(theta)]`.
     """
     if (len(alpha.shape) == 1):
-        return(sp.psi(alpha) - sp.psi(np.sum(alpha)))
-    return(sp.psi(alpha) - sp.psi(np.sum(alpha, 1))[:, np.newaxis])
+        result = psi(alpha) - psi(np.sum(alpha))
+    else:
+        result = psi(alpha) - psi(np.sum(alpha, 1))[:, np.newaxis]
+    return result.astype(alpha.dtype)  # keep the same precision as input
+
+
+def get_random_state(seed):
```

Comment:
Should I instead call it from the `ldamodel` class? How would this rather be done?